source/common/stats/stats_impl.h

```diff
@@ -80,22 +108,31 @@ class CounterImpl : public Counter {
 
   void inc() override { add(1); }
   uint64_t latch() override { return data_.pending_increment_.exchange(0); }
-  std::string name() override { return data_.name_; }
   void reset() override { data_.value_ = 0; }
-  bool used() override { return data_.flags_ & RawStatData::Flags::Used; }
-  uint64_t value() override { return data_.value_; }
+  bool used() const override { return data_.flags_ & RawStatData::Flags::Used; }
+  uint64_t value() const override { return data_.value_; }
+
+  // Stats::Metric
```

Comment:
You can probably avoid duplication here by having a MetricImpl and then doing virtual inheritance. I don't feel that strongly either way.

Reply:
Yeah, I think that makes sense.